### PR TITLE
Simple Javascript Clock (Workshop Bounty)

### PR DESCRIPTION
**Pull request by faisalsayed10**
Criteria:

- [x] Help club members produce a compelling and real project in under 20 minutes
Based on the test meeting, people were able to complete this workshop in under 20 minutes.

- [x] Are very easy to hack on for long periods of time after completion
Many instructions are listed on the end of the workshop as to how people can extend their project and learn more things.

- [x] Have instructive and compelling images, GIFs, and visuals
This workshop do contains instructive and compelling images, GIFs, and visuals.

- [x] Use repl.it or a similar service that allows members to begin coding with one click.
Repl.it is used for this workshop.

- [x] Link unknown concepts to external sources.
Linked many unknown concepts such as Javascript Dates, requestAnimationFrame etc to useful sources from where people can learn more on it.

- [x] End with at least 3 demos of how the project can be expanded, with live links and code links for each demo. 
Have provided 4 demos from the demo workshop to which people can refer and get an idea on how it can be expanded.

Good Workshop Guidelines:

- [x] Workshop that meets all of the "Good workshops" guidelines outlined above.

- [x] 1-2 minute video clip from your test club meeting showing what members built.
Here's the video clip from the demo meeting. People encountered few bugs while doing some custom coding but they got it fixed!
Link: https://hackclub.slack.com/archives/C01A9V5LPD0/p1599927330008900

- [x] All assets (images, PDFs, and any other binaries) must be hosted using the #cdn channel on the Hack Club Slack for posterity and to cut down on repository size.
All assets are hosted using the #cdn channel and gifs have been taken from giphy.

- [x] All workshops must be in English at native-level proficiency. Phrasing and grammar must be perfect.

- [x] You must be either a registered Hack Club leader or an active member of the Slack.
Club leader @ St. Mary's High School and also an active member @ slack.

### CODE REVIEW

**Review by zachlatta — CHANGES_REQUESTED** [review]
Hey @faisalsayed10, this is awesome!

I just did a runthrough of the workshop myself, recording my screen as I did it. I'm currently uploading the video to https://youtu.be/YQ6javQvtK4. It should be live in about 20 minutes.

I think we're really close to merging. I started leaving some comments on GitHub with specific changes I'd like to see before merging, but then I started to just call things out in the video instead of manually commenting for each specific change.

I think we're really close. Here's what I'd like to see before merging:

1. It took me 34 minutes to finish the workshop. Can you shorten it so it only takes me 20? Keep in mind that inside of Hack Club meetings, it'll take members longer to go through the workshop than someone like me doing it alone.
2. There were a number of things, some small (like grammar) and some medium (like some explanations / areas of structure, particularly around the complex CSS and JS stuff), that I ran into in the screen recording. Can you review my screencast and do an opinionated, very detail-oriented edit of the workshop?

Once you've done both of the above, I'll be ready for a final review. This definitely qualifies for the workshop bounty program and I'd love to work with you to get it merged!

**Comment by faisalsayed10** [comment]
@zachlatta Here's what I did:
- [x] Watched the video and made all the necessary changes.
- [x] There was a part where a bug happened when every minute passed, fixed it.
- [x] Fixed the weird whitespaces which appeared in the code blocks. ( github's default tab_size is 8 and that is what is causing it. I've tried to fix as much as I can.
- [x] Added a short explanation + external links for some more complex concepts.

**Comment by zachlatta** [comment]
Awesome @faisalsayed10! @Matt-Gleich is stepping up to help review workshops and manage the Hack Club GitHub repos. His first task: work with you to get this merged! I'm hoping we can get it in over the next few days.

**Comment by Matt-Gleich** [comment]
Hey @faisalsayed10! As Zach said I am going to be helping out with workshop review. I'll check out your PR right now and hopefully, we get it merged ASAP!

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:21]: Change part numbers to normal numbers instead of Roman Numerals.

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:31]:
```suggestion
[Repl.it](https://repl.it) is an online code editor where you can build your clock today. You don't have to use Repl.it but we suggest you do as it sets everything up for you.
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:51]:
```suggestion
This is the `style.css` file in your directory. Whatever styles we give to our website is written in this file. If you take a look in the `<head>` tag in your HTML, you will find a line of code (`<link href="style.css" rel="stylesheet" type="text/css" />`). This means that your HTML file is properly linked to your CSS file!
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:62]:
```suggestion
This means that your HTML file is linked to your JavaScript.
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:77]:
```suggestion
This is what your code should look like so far:
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:304]:
```suggestion
The `Date` object contains specific methods through which we can get the current hour / minute / second.  
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:141]:
```suggestion
Following the 3 main steps given above, let's accomplish this task!
  
1. Select the class designated to the clock div.  
```

File: workshops/Simple Js Clock/README.md
Comment (Matt-Gleich) [inline workshops/Simple Js Clock/README.md:328]:
```suggestion

**Explanation:**
```

**Review by Matt-Gleich — CHANGES_REQUESTED** [review]
Looks great @faisalsayed10! Super approachable, fun, and informative. Just made a few very small change requests and once you implement them we should be good to merge! @zachlatta do you still want to review the latest changes?

**Comment by faisalsayed10** [comment]
@Matt-Gleich I have implemented the changes as you have requested.

**Comment by zachlatta** [comment]
Can you make the folder name all lowercase and use underscores instead of spaces? Make sure you’re happy with the folder name.

From here, if this has your thumbs up @Matt-Gleich, I’d like to do one final run through tomorrow before giving mine and merging.

Congrats! And thanks for all the work you’ve done (Matt you too!).

**Comment by faisalsayed10** [comment]
Yes Sure!

**Review by Matt-Gleich — APPROVED** [review]
Looks good to me! Just fixed that description for ya but you should be good. Congrats on submitting your workshop!

**Comment by Matt-Gleich** [comment]
@zachlatta any update on this?

**Comment by zachlatta** [comment]
Starting my review now!

**Review by zachlatta — CHANGES_REQUESTED** [review]
Hey @faisalsayed10, really good work here. Thank you for all the changes! I just did another workshop runthrough and have a couple final changes for you to make. Please go through it and make all requested changes.

YouTube upload is happening right now. Video at https://youtu.be/OYCciY8Depg. The ETA it's giving me is 1.5 hours until it's online.

I think we are super close. Like 95% there. Really good work on all the revisions since my last review. Once you make these final changes, I think we'll be ready to merge.

I'm going to DM you the link to the workshop payment form so you can get started on that now, since it may take a day or two. Definitely bounty worthy and looking forward to getting you paid!

**Comment by faisalsayed10** [comment]
Hi @zachlatta,
I watched the video and I've done all the changes you mentioned for me to do.
@Matt-Gleich Could you please help me in correcting my grammar at required places? Also, please drop in a final review by your side!

**Comment by Matt-Gleich** [comment]
@faisalsayed10 I will review this as soon as I get out of school today at ~3PM.

**Comment by faisalsayed10** [comment]
Sure

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:3]:
```suggestion
description: 'Build a clock using HTML, CSS, & JavaScript'
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:47]:
```suggestion
At line 1 we have `<!DOCTYPE html>` This declares that this file is an HTML file. If we take a look in the `<html>` tag, we will find a `<body>` tag. Here's where we will write the code. Make sure that all the code you write is included between the opening and closing body tags. (`<body></body>`).
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:72]:
```suggestion
Let us start building our clock! First, we'll write the HTML code required.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:78]:
```suggestion
**Note:** Make sure the `<script>` tag, which is in your body, is below the `<div>` tags, or else it will throw you errors.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:131]:
```suggestion
Then we removed the default margin and padding from all the elements in our website (some elements have a default value for their margin and padding such as 2.4, 3, etc). We changed the default display `display: block` to `display: flex`.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:135]:
```suggestion
Next, we have the `min-height` property which sets a minimum height of `100vh` (viewport height) to your `<body>`. The `min-height` property works together with the `align-items` property and helps the children to align properly at the center.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:168]:
```suggestion
We gave the width and height to our clock. Next, we gave it a border and the radius for the border. Next, is the background color for that clock (if you don't like it, consider changing it or even removing it).
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:179]:
```suggestion
Now let's test our code! Click on the **RUN** button and Voila! Here's the basic background for our clock! (if it doesn't look like this, make sure you crosscheck every single line of code).
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:268]:
```suggestion
**Note:** The `.hand` class was given to 3 `<div>` tags which means this styling will affect all the 3 `<div>` tags. To give some extra classes to the hour hand, minute hand, and the second hand specifically, we also gave those 3 `<div>` tags a unique class name.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:285]:
```suggestion
So far, we have the clock hands aligned correctly and now in this section, we will animate them. Before panicking, let me ensure you that there are just 30 lines of JavaScript code!
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:302]:
```suggestion
Now it's time to implement a function for the proper functionality of the clock. Create a function `setDate` at the bottom of your file.
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:522]:
```suggestion
Here are some things which you can do:
```

File: workshops/simple_clock/README.md
Comment (Matt-Gleich) [inline workshops/simple_clock/README.md:525]:
```suggestion
3. Try to make it have a dark theme!
```

**Review by Matt-Gleich — APPROVED** [review]
Just fix those small little changes and we should be all good!

**Comment by faisalsayed10** [comment]
Fixed all the changes above and I think we should be good to merge?
cc: @zachlatta

**Comment by Matt-Gleich** [comment]
Congrats @faisalsayed10! Your workshop has been merged.

**Comment by faisalsayed10** [comment]
Yay! Thanks!

**Pull request merged**
